clickhouse_search/backend/fields.py

```diff
@@ -15,13 +19,26 @@ def db_type(self, connection):
     def cast_db_type(self, connection):
         return super().cast_db_type(connection).replace('Tuple', 'Nested', 1)
 
+    def from_db_value(self, *args, **kwargs):
+        return self._from_db_value(*args, **kwargs)
+
     def _from_db_value(self, value, expression, connection):
         if value is None:
             return value
-        return [self.container_class(*item) for item in value]
+        value = [self._convert_type(item)._asdict() for item in value]
+        if self.group_key:
+            value = {item[self.group_key]: item for item in value}
```

Comment:
this allows us to write a query against clickhouse that returns an array of tuples, but has it serialized as a dict keyed by an arbitrary object field